crawl-ref/source/webserver/util.py

```diff
@@ -28,6 +28,10 @@ def __init__(self, root_dir):
 
     def load(self, name, parent_path=None):
         name = self.resolve_path(name, parent_path=parent_path)
+        # Fall back to template.html.default if template.html does not exist
```

Comment:
as discussed briefly in ##crawl-dev I want to have a look at how this interacts with dgamelaunch-config's handling of templates (which are copied to the running directory in a somewhat ad-hoc way), but I haven't had a chance to do this yet.

Reply:
Sure. I believe it won't cause problems. The existing paths remain as "server-defined" locations, so dgamelaunch-config's logic will still be correct.

Reply:
Right now, most instances of dgamelaunch-config copy (with overwriting) `client.html` and `game_links.html`, and ignore everything else in that directory. The most recent version (which no one is running yet) copies everything else with an `.html` extension as well but with --no-clobber.

example of typical current setup: https://github.com/neilmoore/dgamelaunch-config/blob/cbro/chroot/sbin/install-trunk.sh#L68
most recent: https://github.com/crawl/dgamelaunch-config/blob/master/chroot/sbin/install-trunk.sh#L68

AFAIK `client.html` should never actually be customized by a server owner, at least in a way that doesn't allow for automatic updates, because this can get modified when changing the webtiles ui -- so I think using this `.default` logic for it is actually wrong (i.e. if it changes in trunk it always needs to get automatically updated). It's possible that someone might customize `game_links.html` (do you know?), though that also seems iffy in the present setup to me. I think my preference would be to not use `.default` versions (i.e. rename them back to the .html version) of these two files so that the current dgamelaunch-config logic will continue to work, and if this logic is needed for game_links.html, phase it in later. And never use it for `client.html`.

The `.default` logic seems ok for the other templates -- it'll require an update to the most recent dgamelaunch to copy those files as well (maybe without `--no-clobber`but that's fine, if someone's already using that they'll be on the ball enough to deal with it. Existing servers using dgamelaunch-config won't need to immediately do anything because they have already copied (and possibly customized) versions in their chroot.

Reply:
Changing the logic to be per-file seems very difficult and brittle (you'd have to hardcode this difference in template loading behaviour). Would you be alright with me adding a `{# comment #}` to the top of client.html saying "don't modify this file, webtiles server admins"?

game_links isn't modified by any server, yet. But I think it will be in the near future for CPO (changing formatting of trunk vs stable vs old stable vs fork).

Reply:
> Changing the logic to be per-file seems very difficult and brittle (you'd have to hardcode this difference in template loading behaviour).

I don't think it's so crazy to hardcode that there are some specific templates that are not overrideable by the server, but maybe what would be simplest for now for compatibility would be to rename `client.html.default` and `game_links.html.default` back to the plain `.html` form and not provide a `.default` version of them in the repository. (I'm not sure if it's intentional, but this is already how `game.html` is handled, which should also not be overrideable and is loaded via this same function.)

Reply:
That's a clever solution!